### PR TITLE
Right click menu

### LINKED ISSUE #30: [Feature] Right click menu
This feature includes a scalable right clicking menu.

### PR DESCRIPTION
**Pull request by BaBrixx**
This closes #30 

### Changes/Additions:
Implemented a dynamic context menu (right click menu).

### CODE REVIEW

**Comment by atjn** [comment]
We just approved a pr (#26) that uses `snake_case` for file names, so we should probably also do this. It is defined in the Google typescript standard: https://google.github.io/styleguide/tsguide.html#redundant-comments

**Comment by NybyDK** [comment]
> We just approved a pr (#26 + #22) that uses `snake_case` for file names, so we should probably also do this. It is defined in the Google typescript standard: https://google.github.io/styleguide/tsguide.html#identifiers-imports

HELLO FROM DENMARK!

Does this also count for .svelte component files or what is up my sir?

**Comment by KamyaPA** [comment]
I'd say that .svelte files should be in PascalCase.svelte

**Comment by atjn** [comment]
> Does this also count for .svelte component files or what is up my sir?

I would say it should count for all files, otherwise it is super confusing to read. That is just my opinion, happy to be corrected.

**Comment by atjn** [comment]
> I'd say that .svelte files should be in PascalCase.svelte

Can you elaborate on why this would be best?

**Comment by BaBrixx** [comment]
I agree with KamyaPA as importing these components would require us to rename them to begin with an uppercase letter:

image

**Comment by KamyaPA** [comment]
When looking at other svelte projects on github that is the most common way of naming .svelte files, and the style they use in the tutorial.

File: src/lib/components/svelte-modifiers/rightClickableComponents/contextMenuItem.ts
Comment (BaBrixx) [inline src/lib/components/svelte-modifiers/rightClickableComponents/contextMenuItem.ts:1]: This file will of course need to be renamed🥌

**Comment by atjn** [comment]
> When looking at other svelte projects on github that is the most common way of naming .svelte files, and the style they use in the tutorial.

I agree with this, which is also why I wanted `.ts` files to be `camelCase`, but we have chosen not to do that. I think it is very confusing to have both naming schemes, so we should choose one or the other. What do you guys think?

**Comment by atjn** [comment]
One of the reasons for using `camelCase` in `.ts` files is that a class `Foo` can also be called `Foo.ts` in the file system. This is what many projects do, and I personally think it's the best way to do it.

**Comment by BaBrixx** [comment]
> One of the reasons for using `camelCase` in `.ts` files is that a class `Foo` can also be called `Foo.ts` in the file system. This is what many projects do, and I personally think it's the best way to do it.

I agree to this for the same reason as for the .svelte files.

**Comment by NybyDK** [comment]
By the power vested in me by the State of AAU, I now pronounce .ts files to be camelCase and .svelte files to be PascalCase
uwu?

**Comment by TSKsmiley** [comment]
PascalCase Pogger 🥌

**Comment by Jesper276** [comment]
a agree with Bastian

**Review by DenFlyvendeGed — DISMISSED** [review]

**Review by atjn — DISMISSED** [review]

**Review by Jesper276 — APPROVED** [review]
LGTM

**Review by atjn — APPROVED** [review]

**Pull request merged**